### PR TITLE
Rename eta_m and eta_m_ref to module_efficiency

### LINKED ISSUE #1188: change eta_m to module_efficiency
`temperature.noct_sam` uses `eta_m_ref` to describe the module efficiency at reference conditions and `temperature.pvsyst_cell` uses `eta_m` to describe the module efficiency generically.

Just calling both of these `module_efficiency` would make the function signatures easily understandable by many more people. I'd be ok with `module_efficiency_ref` but I don't think that precision is very important.

I skimmed [pvterms](https://duramat.github.io/pv-terms/) and didn't see a suggestion for this quantity.

`temperature.noct_sam` has not yet been released and it's just a positional argument, so changing the name is trivial. `temperature.pvsyst_cell` would need a deprecation cycle.

Originally discussed in https://github.com/pvlib/pvlib-python/pull/1177#discussion_r589081257

Assignment of milestone indicates that we will act on this or close it forever before 0.9 is released.

### PR DESCRIPTION
**Pull request by cwhanse**
- [x] Closes #1188
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Tests added
 - [ ] ~~Updates entries to [`docs/sphinx/source/api.rst`](https://github.com/pvlib/pvlib-python/blob/master/docs/sphinx/source/api.rst) for API changes.~~
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

Deprecates `eta_m` in `pvlib.temperature.pvsyst_cell` and replaces `eta_m_ref` in `pvlib.temperature.noct_sam` with `module_efficiency`.

### CODE REVIEW

**Review by mikofski — APPROVED** [review]

**Comment by cwhanse** [comment]
@wholmgren ok to merge?

File: pvlib/temperature.py [hunk 1]
```diff
@@ -313,13 +314,18 @@ def pvsyst_cell(poa_global, temp_air, wind_speed=1.0, u_c=29.0, u_v=0.0,
...
         Combined heat loss factor influenced by wind. Parameter :math:`U_{v}`
         in :eq:`pvsyst`.
-        :math:`\left[ \frac{\text{W}/\text{m}^2}{\text{C}\ \left( \text{m/s} \right)} \right]`
+        :math:`\left[ \frac{\text{W}/\text{m}^2}{\text{C}\ \left( \text{m/s} \right)} \right]`  # noQA: E501
 
-    eta_m : numeric, default 0.1
+    eta_m : numeric, default 0.1 (deprecated, use module_efficiency instead)
         Module external efficiency as a fraction, i.e.,
         :math:`DC\ power / (POA\ irradiance \times module\ area)`.
         Parameter :math:`\eta_{m}` in :eq:`pvsyst`.
```
[the remark this answers is not in the document]
Comment (cwhanse) [reply pvlib/temperature.py]: I agree with all your suggestions.

File: docs/sphinx/source/whatsnew/v0.9.0.rst [hunk 2]
```diff
@@ -67,6 +67,10 @@ Deprecations
   * ``ModelChain.weather``
   * ``ModelChain.times``
 
+* The ``eta_m`` parameter for :py:func:`~pvlib.temperature.pvsyst_cell` is
+  replaced by the ``module_efficiency`` parameter.
```
Comment (wholmgren) [inline docs/sphinx/source/whatsnew/v0.9.0.rst:71]:
```suggestion
  replaced by the ``module_efficiency`` parameter. (:issue:`1188`, :pull:`1218`)
```

**Review by wholmgren — APPROVED** [review]
minor doc suggestion otherwise ready to merge.

**Pull request merged**